### ISSUE TITLE
HELP, React.Exceptions.BabelException: Unable to get value of the property 'split': Object is null or undefined

### ISSUE THREAD

**Issue by vincentshow** [reactjs/React.NET#220]
exception occured when i trying to compile React.Sample.Mvc4 which is downloaded from github.
my ide is vs2012 and install reactjs.net for mvc4(latest version 2.2) by nuget. any suggestion?

**Comment by vincentshow** [comment]
i found that the inner exception occured when JavaScriptEngineSwitcher.Core.JsEngineBase.ExecuteResource processing the resource React.Resources.babel.generated.min.js, and throwed by MsieJavaScriptEngine.ActiveScript. ParseScriptText when calling this._activeScriptParse32.ParseScriptText. 

whatever, no solution yet...

**Comment by alexlblount** [comment]
I'm also receiving this error, but it only happens on the server after I publish.  When I run the project locally, I have no issues.

**Comment by Daniel15** [comment]
Sorry I haven't had time to look into this yet, and I'm about to go on a  vacation from today until 13th February. Could you please try ReactJS.NET 2.1 and see if that works?

**Comment by Daniel15** [comment]
(2.1 is the previous version)

**Comment by rosdi** [comment]
I had the same error.. but when I downgraded to 2.1.2 the problem disappears...

**Comment by Daniel15** [comment]
It's probably due to the upgrade to Babel 6. I'm about to go on vacation
but I can take a look once I'm back. Feel free to use 2.1.2 in the
meantime.

**Comment by Pabloitto** [comment]
I have the same issue, any update ??

**Comment by ajhub** [comment]
@Daniel15 : Hope you had a great vacation.

Same here. Downgrading to 2.1.2 does not help either.
I then get an issue given below ...

React.Exceptions.BabelException: Expected identifier ---> JavaScriptEngineSwitcher.Core.JsRuntimeException: Expected identifier ---> MsieJavaScriptEngine.JsRuntimeException: Expected identifier

Interestingly, there are 5 developers on the project, and this works fine on 3 dev machines. It fails on 2 and also on TeamCity. What makes this machine dependent?
Any help is appreciated.

**Comment by Daniel15** [comment]
@ajhub

> What makes this machine dependent?

By default ReactJS.NET will try to use V8, and fall back to the MSIE JS engine if V8 does not initialise correctly for some reason. There's a few machine-dependent bits:
 - V8 depends on the Visual C++ 2013 runtime (`msvcp120.dll` and `msvcr120.dll`)
 - MSIE depends on the version of Internet Explorer installed

You can disable the MSIE fallback by doing `.SetAllowMsieEngine(false)` in your config.

**Comment by Daniel15** [comment]
I was only able to replicate issues when using the "classic" MSIE engine. This engine is quite old now, so I've totally disabled it in ReactJS.NET and now only allow the newer [Chakra engine](https://en.wikipedia.org/wiki/Chakra_(JScript_engine)).

@vincentshow  - Since you mentioned that you were trying `React.Sample.Mvc4 `, I guess you cloned the ReactJS.NET repo and compiled it? Could you please try that again and see if it works for you now?

**Comment by Daniel15** [comment]
Just pushed ReactJS.NET 2.2.1 which disables the classic MSIE engine as per my comment above. Please reopen this if you still see the same error message.

**Comment by vincentshow** [comment]
note that you mentioned 
MSIE depends on the version of Internet Explorer installed

and i have upgraded my reactjs.net by nuget and added .SetAllowMsieEngine(false) in my config when exception occured,
i try to upgrade my ie from 9 to 11,rebuild......no exception occured and the example was ok!
as i'm in working, there's  no time to downgrade ie to replicate last exception. but this is  interesting, 
am i  missing something rules? doesn't reactjs support ie9?

**Comment by garethdhopkins** [comment]
I had this problem where it worked locally but not on our production server and where adding .SetAllowMsieEngine(false) to my config I got an error about the V8 bin files not being found. Installing [Visual C++ Redistributable Packages for Visual Studio 2013](http://www.microsoft.com/en-us/download/details.aspx?id=40784) fixed the issue.

See the following SO for more information and different fixes:-
http://stackoverflow.com/questions/23301851/clearscript-files-cannot-be-found-on-host

**Comment by milianoo** [comment]
I have this issue on v2.2.1 
"React.Exceptions.BabelException: Unable to get property 'split' of undefined or null reference ---> JavaScriptEngineSwitcher.Core.JsRuntimeException: Unable to get property 'split' of undefined or null reference ---> MsieJavaScriptEngine.JsRuntimeException: Unable to get property 'split' of undefined or null reference "

**Comment by waacton** [comment]
> React.Exceptions.BabelException: Unable to get property 'split' of undefined or null reference

I also had this issue in v2.2.1 after publishing to Azure server.  I disabled MSIE fallback with `.SetAllowMsieEngine(false)` and got:

> DirectoryNotFoundException: Failed to load the ClearScriptV8 assembly, because the directory '........\ClearScript.V8' does not exist

Seems as though v2.2.1 didn't work out of the box for me - I had to copy ClearScript V8 binaries to the right directories (useful: http://www.samulihaverinen.com/web-development/dotnet/2016/01/19/how-to-run-clearscript-v8-javascript-engine-in-azure/).

This is all entirely new to me, so I suspect I've missed something with my environment setup somewhere?
